core/common/src/main/java/alluxio/worker/block/io/BlockReader.java

```diff
@@ -46,4 +48,18 @@
    * @return channel
    */
   ReadableByteChannel getChannel();
+
+  /**
+   * Transfers data from this reader to the buffer.
+   *
+   * @param buf the byte buffer
+   * @return the number of bytes transferred, -1 if the end of the block is reached
+   * @throws IOException if any I/O errors occur
+   */
+  int transferTo(ByteBuf buf) throws IOException;
```

Comment:
any reason to use netty `ByteBuf` rather than java `ByteBuffer` here?
as an interface, should we bind it to netty?